dev/archery/archery/lang/java.py

```diff
@@ -28,3 +31,48 @@ def __init__(self, jar, *args, **kwargs):
         self.jar = jar
         self.argv = ("-jar", jar)
         Java.__init__(self, *args, **kwargs)
+
+
+class JavaConfiguration:
+    def __init__(self,
+
+                 # toolchain
+                 java_home=None, java_options=None,
+                 # build & benchmark
+                 build_extras=None, benchmark_extras=None):
+        self.java_home = java_home
+        self.java_options = java_options
+
+        self.build_extras = build_extras
+        self.benchmark_extras = benchmark_extras
+
+    @property
+    def build_definitions(self):
+        extras = list(self.build_extras) if self.build_extras else []
+        return extras
```

Comment:
I'd recommend doing the cast-to-list here; if an iterator were passed for `build_extras` then it'd be depleted after the first access to `build_definitions`:
```suggestion
        self.build_extras = list(build_extras) if build_extras else []
        self.benchmark_extras = list(benchmark_extras) if benchmark_extras else []

    @property
    def build_definitions(self):
        return self.build_extras
```

I see that CppConfiguration has this same flaw; it's not necessary for you to address it there too